### PR TITLE
fix subset

### PR DESCRIPTION
**Pull request by teghoz**

### CODE REVIEW

File: src/X.PagedList/PagedListExtensions.cs [hunk 1]
```diff
@@ -125,23 +125,20 @@ public static IPagedList<T> ToPagedList<T>(this IQueryable<T> superset, int page
...
-                totalCount = totalSetCount.Value;
-            }
-            else
-            {
-                totalCount = superset.Count();
-            }
+            totalCount = totalSetCount.HasValue ? totalSetCount.Value : superset.Count();
 
-            if (totalCount > 0)
+            if (totalCount > 0 && !totalSetCount.HasValue)
```
[the remark this answers is not in the document]
Comment (teghoz) [reply src/X.PagedList/PagedListExtensions.cs:130]: By default we are taking a subset of the superset. With my previous PR raised and merged, we are extending that approach due to performance issues to take the subset itself. Without that check, it is trying to take a subset of the subset and hence returns and empty collection.

[the remark this answers is not in the document]
Comment (ernado-x) [reply src/X.PagedList/PagedListExtensions.cs:130]: Could you please add unit test for this case?

[the remark this answers is not in the document]
Comment (teghoz) [reply src/X.PagedList/PagedListExtensions.cs:130]: > Could you please add unit test for this case?

Done. Please help merge and update nuget as soon as possible.

File: tests/X.PagedList.Tests/AsyncPagedListExtensionsTheories.cs [hunk 2]
```diff
@@ -34,6 +35,31 @@ public async Task ToListAsync_ForQueryable_WithoutPageNumber_Works(int pageSize,
...
+        var fixture = new Fixture();
+        
+        pageNumber = pageNumber.HasValue == false ? 0 : pageNumber;
+        var listPageNumber = pageNumber != 0 ? pageNumber - 1 : pageNumber;
+        var xListPageNumber = pageNumber == 0 ? 1 : pageNumber;
+        
+        var superset = fixture.CreateMany<Blog>(superSetTotalCount).AsQueryable();
+        var pageOfSuperSet = superset.Skip(listPageNumber.Value * pageSize).Take(pageSize).ToList();
+        
+        var pagedBlogs = await superset.ToPagedListAsync(xListPageNumber, pageSize, superset.Count());
```
Comment (ernado-x) [inline tests/X.PagedList.Tests/AsyncPagedListExtensionsTheories.cs:55]: You changed method _ToPagedList_ but in test you use _ToPagedListAsync_

File: src/X.PagedList/PagedListExtensions.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (teghoz) [reply src/X.PagedList/PagedListExtensions.cs:130]: Hmmm, I see your logic. That is not a problem

File: tests/X.PagedList.Tests/AsyncPagedListExtensionsTheories.cs (hunk 2 above)
Comment (teghoz) [reply tests/X.PagedList.Tests/AsyncPagedListExtensionsTheories.cs:55]: nice spot!

File: src/X.PagedList/PagedListExtensions.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (teghoz) [reply src/X.PagedList/PagedListExtensions.cs:135]: Sorry I was in transit. I am still travelling to my destination—apologies for my slow response. I have added a check for that scenario. When totalSetCount is passed, superSet cannot be greater than totalSetCount. Don't forget the idea is to do the pagination outside the scope of the plugin's logic but still present the pagination regardless.

**Review by ernado-x — APPROVED** [review]

**Pull request merged**